src/daria/analysis/segmentationcomparison.py

```diff
@@ -1,113 +1,471 @@
-    non_active_component: int = kwargs.pop("non_active_component", 0)
-
-    # Define number of segmentations
-    number_of_segmented_images: int = len(segmentations)
-
-    # Define gray colors
-    gray_colors: np.ndarray = kwargs.pop(
-        "gray_colors",
-        np.array([[180, 180, 180], [220, 220, 220], [200, 200, 200]], dtype=np.uint8),
-    )
-    gray_base: np.ndarray = np.fix(gray_colors / number_of_segmented_images).astype(
-        np.uint8
-    )
-
-    # Define unique colors
-    light_scaling: float = kwargs.pop("light_scaling", 1.5)
-    if "colors" not in kwargs:
-        colormap = get_cmap("Spectral")
-        colors: np.ndarray = np.zeros(
-            (number_of_segmented_images, 2, 3), dtype=np.uint8
+
+    def __init__(self, number_of_segmented_images: int, **kwargs) -> None:
+        """
+        Constructor of compare segmentations class.
+
+        Args:
+            number_of_segmented_images (int): Number of segmentations to be compared
+            Optional keyword arguments (kwargs):
+                segmentation_names (list): list of names for the different segmented
+                    images. So far only used in legends and color dictionary.
+                components (list): list of the different components that are
+                    considered in the segmented images. So far only two are allowed
+                    to be provided.
+                component_names (list):  list of names for the different components.
+                    So far only used in legends, and color dictionary.
+                non_active_component (int): value of nonactive component in the segmented
+                    images.
+                gray_colors (np.ndarray): array of three different scales of
+                    gray (in RGB format), one for each of the different combinations of
+                    components in the segmentations.
+                colors (np.ndarray): Array of different colors that should
+                    indicate unique components in each segmentation.
+                light_scaling (float): Indicate how much lighter the second
+                    component should be scaled in its unique color.
+        """
+
+        self.number_of_segmented_images = number_of_segmented_images
+        self.segmentation_names: list = kwargs.pop(
+            "segmentation_names",
+            [f"Segmentation {i}" for i in range(self.number_of_segmented_images)],
+        )
+
+        # Define components
+        self.components: list = kwargs.pop("components", [1, 2])
+        self.non_active_component: int = kwargs.pop("non_active_component", 0)
+        self.component_names: list = kwargs.pop(
+            "component_names", ["Component 0", "Component 1"]
+        )
+
+        # Define gray colors
+        self.gray_colors: np.ndarray = kwargs.pop(
+            "gray_colors",
+            np.array(
+                [[180, 180, 180], [220, 220, 220], [200, 200, 200]], dtype=np.uint8
+            ),
```

Comment:
If really only gray colors are intended, wouldn't it be better to only ask for monochromatic values instead of trichromatic?

Reply:
I thought about it, but decided that probably it could be a good idea to have the option to have other colors than gray in the future when more segmentations are compared. Moreover, I am not sure how easy it is to combine monochromatic and rgb colors in the same image?